### PR TITLE
remove all Problems from fcrepo-kernel-api

### PR DESCRIPTION
**Pull request by acoburn**
Resolves: https://jira.duraspace.org/browse/FCREPO-1834

### CODE REVIEW

File: fcrepo-jms/pom.xml [hunk 1]
```diff
@@ -47,6 +47,10 @@
       <artifactId>javax.inject</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.modeshape</groupId>
```
Comment (awoods) [inline fcrepo-jms/pom.xml]: Do you see what requires `org.modeshape` in the `fcrepo-jms` module? It seems like that is the next target.

Comment (acoburn) [inline fcrepo-jms/pom.xml]: Yes, it is the JCR_CONTENT constant. It implies a deeper investigation as to why it is escaping the fcrepo-kernel-modeshape module. And yes, it is most certainly a good "next target".

Comment (acoburn) [inline fcrepo-jms/pom.xml]: Also, I suspect that finding where the JCR_CONTENT value is escaping from fcrepo-kernel-modeshape will also help remove that from fcrepo-module-auth-webac: http://git.io/v854c

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/repository/FedoraRepositoryBackup.java [hunk 2]
```diff
@@ -90,21 +90,16 @@ public String runBackup(final InputStream bodyStream) throws IOException {
...
-            final StringBuilder problemsOutput = new StringBuilder();
-
             // Report the problems (we'll just print them out) ...
-            for ( final Problem problem : problems ) {
-                LOGGER.error("{}", problem.getMessage());
-                problemsOutput.append(problem.getMessage());
-                problemsOutput.append("\n");
-            }
+            final String output = problems.stream().map(Throwable::getMessage).peek(x -> LOGGER.error("{}", x))
+                    .collect(Collectors.joining("\n"));
```
Comment (ajs6f) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/repository/FedoraRepositoryBackup.java]: Static import `joining` and you will make me smile.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/repository/FedoraRepositoryBackup.java [hunk 3]
```diff
@@ -90,21 +90,16 @@ public String runBackup(final InputStream bodyStream) throws IOException {
...
 
-            final StringBuilder problemsOutput = new StringBuilder();
-
             // Report the problems (we'll just print them out) ...
-            for ( final Problem problem : problems ) {
-                LOGGER.error("{}", problem.getMessage());
-                problemsOutput.append(problem.getMessage());
-                problemsOutput.append("\n");
-            }
+            final String output = problems.stream().map(Throwable::getMessage).peek(x -> LOGGER.error("{}", x))
```
Comment (ajs6f) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/repository/FedoraRepositoryBackup.java]: Why not just `peek(LOGGER::error)`?

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/RepositoryServiceImpl.java [hunk 4]
```diff
@@ -97,14 +99,20 @@ public Long getRepositoryObjectCount() {
...
                                      final File backupDirectory) {
         try {
             final RepositoryManager repoMgr = ((org.modeshape.jcr.api.Session) session)
                     .getWorkspace()
                     .getRepositoryManager();
 
-            return repoMgr.backupRepository(backupDirectory);
+            final Collection<Throwable> problems = new ArrayList<>();
+
+            repoMgr.backupRepository(backupDirectory).iterator().forEachRemaining(x -> {
```
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/RepositoryServiceImpl.java]: Why not just `.backupRepository(backupDirectory)::forEach`?

File: fcrepo-jms/pom.xml (hunk 1 above)
Comment (awoods) [inline fcrepo-jms/pom.xml]: :+1:

**Comment by acoburn** [comment]
@ajs6f updated PR with your suggestions

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/RepositoryServiceImpl.java [hunk 5]
```diff
@@ -97,14 +99,20 @@ public Long getRepositoryObjectCount() {
...
                                      final File backupDirectory) {
         try {
             final RepositoryManager repoMgr = ((org.modeshape.jcr.api.Session) session)
                     .getWorkspace()
                     .getRepositoryManager();
 
-            return repoMgr.backupRepository(backupDirectory);
+            final Collection<Throwable> problems = new ArrayList<>();
+
+            repoMgr.backupRepository(backupDirectory).forEach(x -> {
```
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/RepositoryServiceImpl.java]: No need for braces and newlines here, just `.backupRepository(backupDirectory).forEach(x -> problems.add(x.getThrowable()));`

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo4/commit/89b8bff3b976c2116b38e0409395ab75f64caaab

**Pull request closed without merging**